org.envirocar.app/src/org/envirocar/app/services/trackchunks/TrackchunkUploadService.java

```diff
@@ -165,7 +190,13 @@ public void onReceiveNewMeasurementEvent(RecordingNewMeasurementEvent event) {
         measurements.add(event.mMeasurement);
         LOG.info("received new measurement" + this);
         if(measurements.size() > MEASUREMENT_THRESHOLD && currentTrack != null && currentTrack.getRemoteID() != null) {
-           List<Measurement> measurementsCopy = new ArrayList<>(measurements.size() + 1);
+            if (currentTrack.getMeasurements().size() == 1) {
+                currentTrack.addMeasurements(measurements.subList(1, measurements.size()));
+            }
+            else {
+                currentTrack.addMeasurements(measurements);
+            }
+            List<Measurement> measurementsCopy = new ArrayList<>(measurements.size() + 1);
```

Comment:
We have to add the buffered measurements to the `currentTrack` so that, when track will be finished, `getDistanceOfTrack()` can be called on it, which calculates the length of the track based on the measurements. The previous implementation only added the first measurement to the `currentTrack`, which causes that no length could be calculated any more after the initial upload.

Reply:
Sounds reasonable. thanks!